core/src/main/java/io/questdb/griffin/SqlCompiler.java

```diff
@@ -515,8 +515,34 @@ private CompiledQuery alterTable(SqlExecutionContext executionContext) throws Sq
                     } else {
                         throw SqlException.$(lexer.lastTokenPosition(), "'param' expected");
                     }
+                } else if (SqlKeywords.isResumeKeyword(tok)) {
+                    tok = expectToken(lexer, "'wal'");
+                    if (!SqlKeywords.isWalKeyword(tok)) {
+                        throw SqlException.$(lexer.lastTokenPosition(), "'wal' expected");
+                    }
+
+                    tok = SqlUtil.fetchNext(lexer); // optional from part
+                    long fromTxn = -1;
+                    if (tok != null) {
+                        if (SqlKeywords.isFromKeyword(tok)) {
+                            tok = expectToken(lexer, "'transaction' or 'txn'");
+                            if (!(SqlKeywords.isTransactionKeyword(tok) || SqlKeywords.isTxnKeyword(tok))) {
+                                throw SqlException.$(lexer.lastTokenPosition(), "'transaction' or 'txn' expected");
+                            }
+                            CharSequence txnValue = expectToken(lexer, "transaction value");
+                            final int valuePosition = lexer.getPosition();
+                            try {
+                                fromTxn = Numbers.parseLong(txnValue);
+                            } catch (NumericException e) {
+                                throw SqlException.$(valuePosition, "invalid value [value=").put(txnValue).put(']');
+                            }
+                        } else {
+                            throw SqlException.$(lexer.lastTokenPosition(), "'from' expected");
+                        }
+                    }
+                    return alterTableResume(tableNamePosition, tableName, fromTxn, executionContext);
                 } else {
-                    throw SqlException.$(lexer.lastTokenPosition(), "'add', 'drop', 'attach', 'detach', 'set' or 'rename' expected");
+                    throw SqlException.$(lexer.lastTokenPosition(), "'add', 'drop', 'attach', 'detach', 'set', 'rename' or 'resume' expected");
                 }
             } catch (CairoException e) {
                 LOG.info().$("could not alter table [table=").$(tableName).$(", ex=").$((Throwable) e).$();
```

Comment:
i know this is not new, but re-shaping exception is no longer necessary here. `CairoException` now has `position`, e.g.
```
e.position(lexer.lastTokenPosition());
throw e;
```

should suffice.